pytorch_toolkit/ote/ote/modules/config_transformers/nncf_config_transformer.py

```diff
@@ -0,0 +1,115 @@
+            return {}
+        assert isinstance(compression_template, dict), f'Error: compression part of template is not a dict: template["compression"]={compression_template}'
+        possible_keys = set(self.POSSIBLE_NNCF_PARTS) | set([self.COMPRESSION_CONFIG_KEY])
+        unknown_keys = set(compression_template.keys()) - set(possible_keys)
+        if unknown_keys:
+            raise RuntimeError(f'Compression parameters contain unknown keys: {list(unknown_keys)}')
+
+        compression_template = copy(compression_template)
+        if self.COMPRESSION_CONFIG_KEY not in compression_template:
+            raise RuntimeError(f'Error: compression part of template does not contain the key {self.COMPRESSION_CONFIG_KEY}')
+        compression_config_path = compression_template.pop(self.COMPRESSION_CONFIG_KEY)
+
+        compression_parts_to_choose = []
+        for k, v in compression_template.items():
+            should_pick = _convert_value_to_bool(v, f'compression.{k}')
+            if should_pick:
+                compression_parts_to_choose.append(k)
+
+        if not compression_parts_to_choose:
+            return {}
+
+        nncf_config_part = self._merge_nncf_compression_parts(compression_config_path, compression_parts_to_choose)
+        return nncf_config_part
+
+    @staticmethod
+    def _merge_nncf_compression_parts(compression_config_path, compression_parts_to_choose):
+        compression_parts = Config.fromfile(compression_config_path)
+
+        if 'order_of_parts' in compression_parts:
+            order_of_parts = compression_parts['order_of_parts']
+            assert isinstance(order_of_parts, list), 'The field "order_of_parts" in compression config should be a list'
+
+            for part in compression_parts_to_choose:
+                assert part in order_of_parts, (
+                        f'The part {part} is selected, but it is absent in order_of_parts={order_of_parts},'
+                        f' see the compression config file {compression_config_path}')
+
+            compression_parts_to_choose = [part for part in order_of_parts if part in compression_parts_to_choose]
+
+        assert 'base' in compression_parts, f'Error: the compression config does not contain the "base" part'
+        nncf_config_part = compression_parts['base'].to_dict()
+
+        for part in compression_parts_to_choose:
+            assert part in compression_parts, (
+                    f'Error: the compression config does not contain the part "{part}", '
+                    f'whereas it was selected; see the compression config file "{compression_config_path}"')
+            compression_part_dict = compression_parts[part].to_dict()
+            try:
+                nncf_config_part = merge_dicts_and_lists_b_into_a(nncf_config_part, compression_part_dict)
+                cur_error = None
+            except AssertionError as e:
+                cur_error = e
+
+            if cur_error:
```

Comment:
Because during merging different structs in some cases the merging function throws exception.

The exception describes what is wrong with merging.

It happens e.g. if we merge `{'a': {'b': 1} }` with `{'a': [1,2,3]}` -- merging of lists and dicts is impossible

See the test cases at the bottom of the file
`pytorch_toolkit/ote/ote/modules/config_transformers/utils.py`

Reply:
That was clear. I see no need for an extra variable. Just throw your error from the `except` section.

Reply:
As far as I know, if move raising another exception inside `except`, then python3 from some version (may be >= 3.7, I'm not sure) prints that 'exception has happened during handling of another exception', etc.

I do not want to print this stuff ('one exception was happened inside handling of another exception', etc), I just want to add a bit more descriptive lines before the inner exception's description. So, I convert the inner exception to string and raise a new one with this string inside.

Reply:
Ok, intention is clear now. But I would still prefer to avoid that kind of workarounds.

`raise RuntimeError('error message') from None`

should do the job even being called from within the `except` clause, please correct me if it doesn't.

Reply:
Oh! cool trick, I did not know it.

Made a commit with this change. Done.